### PR TITLE
feat: Stargaze spec from Impulse

### PR DESCRIPTION
**Pull request by MichaelLLC**
Hello, I'm Michael, founder of Impulse.expert, specializing in high-performance, secure, and scalable node operations across 21 chains, maintaining 99% uptime. I've been actively involved with Lava since October 21, 2022, contributing to docs, socials and automation scripts. I'm deeply engaged in community support and project development. The custom automation and monitoring systems ensure reliable operations. I'm eager to maintain the Stargaze RPC spec on Lava and continue supporting the community with my expertise.

Commission: 2.5%:

    1% - Support and run a provider
    0.5% - Inherit, maintain and update Stargaze spec
    1% - Provider support

Discord: michaels1473

### CODE REVIEW

File: mainnet-1/specs/stargaze.json [hunk 1]
```diff
@@ -0,0 +1,780 @@
...
+    "proposal": {
+        "title": "Add Specs: Stargaze",
+        "description": "Adding new specification support for relaying agoric data on stargaze",
+        "specs": [
+            {
+                "index": "STRGZ",
+                "name": "stargaze mainnet",
+                "enabled": true,
+                "imports": [
+                    "COSMOSSDKFULL"
```
Comment (Yaroms) [inline mainnet-1/specs/stargaze.json:11]: stargaze is using cosmossdk v50
the specs needs to import from cosmossdk50

Comment (MichaelLLC) [reply mainnet-1/specs/stargaze.json]: Fix it

**Comment by Yaroms** [comment]
all good now i approve

**Comment by Yaroms** [comment]
please make sure you are supporting only relevant apis:
rest:
```
"/stargaze/tokenfactory/v1/denoms_from_creator/{creator}"
"/stargaze/globalfee/v1/params"
"/stargaze/globalfee/v1/authorizations"
"/stargaze/cron/v1/list-privileged"
"/stargaze/alloc/v1beta1/params"
"/stargaze/globalfee/v1/code_authorization/{code_id}"
"/stargaze/tokenfactory/v1/params"
"/stargaze/tokenfactory/v1/denoms/{denom}/authority_metadata"
"/stargaze/mint/v1beta1/annual_provisions"
"/stargaze/globalfee/v1/contract_authorization/{contract_address}"
"/stargaze/mint/v1beta1/params"
"/stargaze/cron/v1/params"
```

grpc:
```
"publicawesome.stargaze.cron.v1.Query/ListPrivileged"
"osmosis.tokenfactory.v1beta1.Query/DenomsFromCreator"
"publicawesome.stargaze.globalfee.v1.Query/CodeAuthorization"
"publicawesome.stargaze.mint.v1beta1.Query/AnnualProvisions"
"publicawesome.stargaze.globalfee.v1.Query/ContractAuthorization"
"publicawesome.stargaze.mint.v1beta1.Query/Params"
"osmosis.tokenfactory.v1beta1.Query/DenomAuthorityMetadata"
"publicawesome.stargaze.globalfee.v1.Query/Params"
"publicawesome.stargaze.cron.v1.Query/Params"
"publicawesome.stargaze.globalfee.v1.Query/Authorizations"
"publicawesome.stargaze.alloc.v1beta1.Query/Params"
"osmosis.tokenfactory.v1beta1.Query/Params"
```

**Comment by Yaroms** [comment]
please make sure to support the updated api list:

rest:```

"/stargaze/cron/v1/params"
"/stargaze/tokenfactory/v1/denoms/{denom}/authority_metadata"
"/stargaze/alloc/v1beta1/params"
"/stargaze/mint/v1beta1/params"
"/stargaze/mint/v1beta1/annual_provisions"
"/stargaze/tokenfactory/v1/params"
"/stargaze/tokenfactory/v1/denoms_from_creator/{creator}"
"/stargaze/globalfee/v1/contract_authorization/{contract_address}"
"/stargaze/globalfee/v1/code_authorization/{code_id}"
"/stargaze/cron/v1/list-privileged"
"/stargaze/globalfee/v1/params"
"/stargaze/globalfee/v1/authorizations"
```

grpc:
```

"publicawesome.stargaze.globalfee.v1.Query/Params"
"publicawesome.stargaze.mint.v1beta1.Query/AnnualProvisions"
"osmosis.tokenfactory.v1beta1.Query/Params"
"publicawesome.stargaze.alloc.v1beta1.Query/Params"
"publicawesome.stargaze.globalfee.v1.Query/Authorizations"
"publicawesome.stargaze.cron.v1.Query/ListPrivileged"
"publicawesome.stargaze.cron.v1.Query/Params"
"publicawesome.stargaze.mint.v1beta1.Query/Params"
"publicawesome.stargaze.globalfee.v1.Query/CodeAuthorization"
"publicawesome.stargaze.globalfee.v1.Query/ContractAuthorization"
"osmosis.tokenfactory.v1beta1.Query/DenomsFromCreator"
"osmosis.tokenfactory.v1beta1.Query/DenomAuthorityMetadata"

```

**Comment by MichaelLLC** [comment]
I checked the specification. All APIs are present in the specification.

**Pull request closed without merging**